extensions/positron-r/src/kernel-spec.ts

```diff
@@ -0,0 +1,169 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as positron from 'positron';
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+
+import { JupyterKernelSpec } from './jupyter-adapter';
+import { getArkKernelPath } from './kernel';
+import { getPandocPath } from './pandoc';
+import { EXTENSION_ROOT_DIR } from './constants';
+
+/**
+ * Create a new Jupyter kernel spec.
+ *
+ * @param rHomePath The R_HOME path for the R version
+ * @param runtimeName The (display) name of the runtime
+ * @param sessionMode The mode in which to create the session
+ *
+ * @returns A JupyterKernelSpec definining the kernel's path, arguments, and
+ *  metadata.
+ */
+export function createJupyterKernelSpec(
+	rHomePath: string,
+	runtimeName: string,
+	sessionMode: positron.LanguageRuntimeSessionMode): JupyterKernelSpec {
+
+	// Path to the kernel executable
+	const kernelPath = getArkKernelPath();
+	if (!kernelPath) {
+		throw new Error('Unable to find R kernel');
+	}
+
+	// Check the R kernel log level setting
+	const config = vscode.workspace.getConfiguration('positron.r');
+	const logLevel = config.get<string>('kernel.logLevel') ?? 'warn';
+	const logLevelForeign = config.get<string>('kernel.logLevelExternal') ?? 'warn';
+	const userEnv = config.get<object>('kernel.env') ?? {};
+	const profile = config.get<string>('kernel.profile');
+
+
+	/* eslint-disable */
+	const env = <Record<string, string>>{
+		'RUST_BACKTRACE': '1',
+		'RUST_LOG': logLevelForeign + ',ark=' + logLevel,
+		'R_HOME': rHomePath,
+		...userEnv
+	};
```

Comment:
Minor observation: we are using a little interface for env vars elsewhere in positron-r (but I don't think this matters that much and I see this code pre-existed, it's just moving in this PR):

https://github.com/posit-dev/positron/blob/1492e58b74678103a84810ae453cfa65cc4662ad/extensions/positron-r/src/session.ts#L30-L32

https://github.com/posit-dev/positron/blob/1492e58b74678103a84810ae453cfa65cc4662ad/extensions/positron-r/src/uri-handler.ts#L69

https://github.com/posit-dev/positron/blob/1492e58b74678103a84810ae453cfa65cc4662ad/extensions/positron-r/src/session.ts#L927

Reply:
Good observation! Indeed I didn't touch this (just moved it to keep the file from becoming too unwieldy) but that'd be an improvement.